exercises/complex-numbers/complex_numbers_test.cpp

```diff
@@ -0,0 +1,218 @@
+#include "complex_numbers.h"
+
+#include <cmath>
+#include <complex>
+
+#include "test/catch.hpp"
+
+// Define M_PI for platforms that don't include it
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+// Complex Numbers exercise test case data version 1.0.0
+
+using complex_numbers::Complex;
+
+TEST_CASE("add_purely_real_numbers") {
+    const Complex c1{1.0, 0.0};
+    const Complex c2{2.0, 0.0};
+
+    REQUIRE(Complex(3.0, 0.0) == c1 + c2);
+}
+
+#if defined(EXERCISM_RUN_ALL_TESTS)
+TEST_CASE("add_purely_imaginary_numbers") {
+    const Complex c1{0.0, 1.0};
+    const Complex c2{0.0, 2.0};
+
+    REQUIRE(Complex(0.0, 3.0) == c1 + c2);
+}
+
+TEST_CASE("add_numbers_with_real_and_imaginary_part") {
+    const Complex c1{1.0, 2.0};
+    const Complex c2{3.0, 4.0};
+
+    REQUIRE(Complex(4.0, 6.0) == c1 + c2);
+}
+
+TEST_CASE("subtract_purely_real_numbers") {
+    const Complex c1{1.0, 0.0};
+    const Complex c2{2.0, 0.0};
+
+    REQUIRE(Complex(-1.0, 0.0) == c1 - c2);
+}
+
+TEST_CASE("subtract_purely_imaginary_numbers") {
+    const Complex c1{0.0, 1.0};
+    const Complex c2{0.0, 2.0};
+
+    REQUIRE(Complex(0.0, -1.0) == c1 - c2);
+}
+
+TEST_CASE("subtract_numbers_with_real_and_imaginary_part") {
+    const Complex c1{1.0, 2.0};
+    const Complex c2{3.0, 4.0};
+
+    REQUIRE(Complex(-2.0, -2.0) == c1 - c2);
+}
+
+TEST_CASE("multiply_purely_real_numbers") {
+    const Complex c1{1.0, 0.0};
+    const Complex c2{2.0, 0.0};
+
+    REQUIRE(Complex(2.0, 0.0) == c1 * c2);
+}
+
+TEST_CASE("multiply_purely_imaginary_numbers") {
+    const Complex c1{0.0, 1.0};
+    const Complex c2{0.0, 2.0};
+
+    REQUIRE(Complex(-2.0, 0.0) == c1 * c2);
+}
+
+TEST_CASE("multiply_numbers_with_real_and_imaginary_part") {
+    const Complex c1{1.0, 2.0};
+    const Complex c2{3.0, 4.0};
+
+    REQUIRE(Complex(-5.0, 10.0) == c1 * c2);
+}
+
+TEST_CASE("imaginary_unit") {
```

Comment:
The reordered problem specifications repo currently puts the test for [imaginary unit](https://github.com/chgraef/cpp/blob/7cad6c17cb12d39ef888745b32fb7755de284291/exercises/complex-numbers/complex_numbers_test.cpp#L81) in a weird place. I think it should stay with the multiplication examples as we have it here. I've opened [an issue](https://github.com/exercism/problem-specifications/issues/1584) in the problem specs repo.

Reply:
That's fair. I've updated to use the latest version in the problem specifications repo, though we could wait to see if the decision is made to change the tests before merging this in.

Reply:
I'm a little more inclined to merge this as soon as we're done. Apparently the exercise ordering hasn't been a big problem for students/mentors? And I'll get a notification if the problem spec changes.

Reply:
:+1: